src/components/AuthenticatedContributeMeasurementForm.tsx

```diff
@@ -147,7 +149,30 @@ export const AuthenticatedForm: FC<AuthenticatedFormProps> = (props) => {
     undefined
   );
 
+  const checkIfNodeValid = async() => {
+    if (nodeId !== "") {
+      const isValid = await isNodeValid(nodeId);
+      setIsSuppliedNodeValid(isValid);
+
+    } else {
+      setIsSuppliedNodeValid(undefined);
```

Comment:
If isSuppliedNodeId is undefined by default we can leave this call out?

Reply:
Done.